src/astro/electromagnetism/radiationPressureTargetModel.cpp

```diff
@@ -82,51 +83,88 @@ void PaneledRadiationPressureTargetModel::updateRadiationPressureForcing( double
                 throw std::runtime_error( "Torques not yet supported for moving vehicle parts." );
             }
         }
-
         const std::vector< std::shared_ptr< system_models::VehicleExteriorPanel > >& currentPanels_ =
                 ( i == 0 ) ? bodyFixedPanels_ : segmentFixedPanels_.at( segmentFixedPanelsIterator->first );
+        
         for( unsigned int j = 0; j < currentPanels_.size( ); j++ )
+        {   
+            allRotatedPanels_[ buffer + j ] = std::make_shared< system_models::VehicleExteriorPanel >( currentPanels_.at( j ),
+                                                                                            currentOrientation );
+        }
+        buffer += currentPanels_.size( );
+        if( i > 0 )
         {
-            surfaceNormals_[ counter ] = currentOrientation * currentPanels_.at( j )->getFrameFixedSurfaceNormal( )( );
-            surfacePanelCosines_[ counter ] = ( -sourceToTargetDirectionLocalFrame ).dot( surfaceNormals_[ counter ] );
+            segmentFixedPanelsIterator++;
+        }
+    }
+    // SSH off
+    if ( maximumNumberOfPixels_ == 0 )
+    {
+        illuminatedPanelFractions_ = std::vector< double >( totalNumberOfPanels_, 1.0);
+    }
+    // SSH on
+    if ( maximumNumberOfPixels_ < 0 )
+    {
+        throw std::runtime_error("Error: maximum number of pixels must be positive and at least 2!");
+    }
+    if ( maximumNumberOfPixels_ == 1 )
+    {
+        if ( macroModelLoaded_ )
+        {
+            throw std::runtime_error("Error: maximum number of pixels must be at least two!");
+        }
+        if ( !macroModelLoaded_ )
+        {
+            throw std::runtime_error("Error: maximum number of pixels must be at least two and macro-model is not loaded!");
+        }
+    }
+    if ( maximumNumberOfPixels_ >= 2 )
+    {
+        if ( !macroModelLoaded_ )
+        {
+            throw std::runtime_error("Error: maximum number of pixels given but macro-model is not loaded!");
+        }
+        // compute illuminated fractions with SSH algorithm
+        selfShadowing_ = system_models::SelfShadowing( allRotatedPanels_, sourceToTargetDirectionLocalFrame, maximumNumberOfPixels_ );
```

Comment:
Rather than re-creating the object every time, I'd recommend modifying this so that the same selfShadowing_ object is used in every call. Creating/destroying the object every time can incur significant memory overhead